chrome/content/zotero/xpcom/itemTreeManager.js

```diff
@@ -157,16 +157,29 @@ class ItemTreeManager {
 			return allColumns;
 		}
 		let filteredColumns = allColumns;
-		if (enabledTreeIDs) {
-			if (typeof enabledTreeIDs === "string") {
-				enabledTreeIDs = [enabledTreeIDs];
+		if (typeof enabledTreeIDs === "string") {
+			enabledTreeIDs = [enabledTreeIDs];
+		}
+		// If enabledTreeIDs is specified, filter columns by enabledTreeIDs
+		// If enabledTree is "*", should not be filtered
+		if (enabledTreeIDs && !enabledTreeIDs.includes("*")) {
+			const enabledTreeIDsSet = new Set(enabledTreeIDs);
+
+			/**
+			 * Check if the column is enabled for any of the specified trees
+			 * @param {string[]} colEnabledTreeIDs - The tree IDs the column is enabled for
+			 * @returns {boolean} true if the column is enabled for any of the specified trees
+			 */
+			// eslint-disable-next-line no-inner-declarations
+			function hasEnabledTreeID(colEnabledTreeIDs) {
+				if (colEnabledTreeIDs.includes("*")) {
+					return true;
+				}
+				// See https://stackoverflow.com/a/43820518
+				// The most efficient solution for small arrays: find + has (arrow function)
+				return colEnabledTreeIDs.find(treeID => enabledTreeIDsSet.has(treeID));
 			}
-			filteredColumns = filteredColumns.filter((col) => {
-				const colEnabledTreeIDs = col.enabledTreeIDs || ["main"];
-				return colEnabledTreeIDs.includes("*")
-					|| enabledTreeIDs.includes("*")
-					|| enabledTreeIDs.find(treeID => colEnabledTreeIDs.includes(treeID));
-			});
+			filteredColumns = filteredColumns.filter(col => hasEnabledTreeID(col.enabledTreeIDs));
```

Comment:
I am sorry I am being so asinine about this part. This may just be me not being used to reading and writing this kind of code. I actually sat down and wrote this in a way that was most readable to me just to see if I'm just asking for impossible or what. This is what I got:

```js
filterTreeIDs = new Set(filterTreeIDs);
filteredColumns = filteredColumns.filter(column => {
	if (column.enabledTreeIDs[0] == "*") return true;

	for (treeID of column.enabledTreeIDs) {
		if (filterTreeIDs.has(treeID)) return true;
	}
})
```

This may not be as efficient in terms of CPU cycles as using Array.find(), but I find it much more readable, because you can clearly see under what conditions we filter true. Also it's probably not an issue, but technically if we had a tree with id "0" the find would run into a corner case here, so Array.some() would be more appropriate.

Moreover, when I originally requested changes to this, I suggested that this function should not take array arguments at all, i.e. just single arguments would suffice. That would also have made this filter simpler.

You've expanded this piece of code a lot and added an intermediate function and a lot of comments. Comments and restructuring are the last resort in making something clear and sometimes unavoidable for really complex code. I guess you went this way because you did not know what I'm looking for, but I don't think this verbosity is good either in making things clearer. Sorry!

I'm OK if you think that the nested comprehension is just as readable (or more readable) as mine and go back to an earlier version that is less verbose, with just an inline function, or you can just take my code above.